src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,41 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.BulldozerProducer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.ExcavatorProducer;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.TruckProducer;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl<T extends Machine> implements MachineService<T> {
```

Comment:
we should move from abstraction to implementation. So we don't need to parameterize MachineServiceImpl

Reply:
On the lecture, Bohdan was implementing MachineServiceImpl and made it as I did. Then he explained that this is better than non-parametrized MachineServiceImpl cause we achieve better flexibility. I tend to leave my solution there